### PR TITLE
Add tailwind-merge in combinaison with clsx

### PR DESCRIPTION
**Pull request by sebastien-comeau**
### Description

A function that allows the conditional classnames from ‘clsx’ or ‘classnames’ to be passed into ‘tailwind-merge’.

Combining [clsx](https://www.npmjs.com/package/clsx) or [classnames](https://www.npmjs.com/package/classnames) with [tailwind-merge](https://www.npmjs.com/package/tailwind-merge) allows us to conditionally join [Tailwind CSS](https://tailwindcss.com/) classes in classNames together without style conflicts. Inspired by [shadcn/ui](https://ui.shadcn.com/).

### Related Azure Boards Work Items

_N/A_

### Screenshots (if applicable)

_N/A_

### Checklist

- [x]  I have tested the changes locally
- [x] I have updated the documentation if necessary
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [x] I have checked that all e2e tests pass by running `npm run test:e2e`

### Test Instructions

_N/A_

### Additional Notes

_N/A_

### CODE REVIEW

**Review by gregory-j-baker — DISMISSED** [review]

File: frontend/app/utils/tw-utils.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+/**
+ * A function that allows the conditional classnames from 'clsx' or 'classnames' to be passed into 'tailwind-merge'.
+ *
+ * Combining clsx or classnames with tailwind-merge allows us to conditionally join Tailwind CSS classes in classNames
+ * together without style conflicts. Inspired by [shadcn/ui](https://ui.shadcn.com/).
+ *
+ * @param inputs - Class names or objects representing class names to be merged.
+ * @returns The merged class names.
+ */
+export function cn(...args: ClassValue[]) {
```
[the remark this answers is not in the document]
Comment (sebastien-comeau) [reply frontend/app/utils/tw-utils.ts:13]: `cn` is an abbreviation for `className`

[the remark this answers is not in the document]
Comment (gregory-j-baker) [reply frontend/app/utils/tw-utils.ts:13]: Doesn't matter.. `cn()` is a bad name for a function.

[the remark this answers is not in the document]
Comment (sebastien-comeau) [reply frontend/app/utils/tw-utils.ts:13]: Any name suggestion?

[the remark this answers is not in the document]
Comment (sebastien-comeau) [reply frontend/app/utils/tw-utils.ts:13]: It's a name used with those 2 libraries when mixed together.
https://medium.com/@gorkemkaramolla/react-tailwind-reuseable-and-customizable-components-with-cva-clsx-and-tailwindmerge-combo-guide-c3756bdbbf16#553f

[the remark this answers is not in the document]
Comment (sebastien-comeau) [reply frontend/app/utils/tw-utils.ts:13]: [cn() - Every Tailwind Coder Needs It (clsx + twMerge)](https://www.youtube.com/watch?v=re2JFITR7TI)

[the remark this answers is not in the document]
Comment (nicholas-ly) [reply frontend/app/utils/tw-utils.ts:13]: I agree that `cn` may not be obvious but it seems to be a Shadcn convention? If so, I am okay with it. 

It's close but `cn` > `clsx` to me.

**Review by nicholas-ly — APPROVED** [review]

**Comment by gregory-j-baker** [comment]
Noooooooooooooooooooooooo!

**Pull request merged**